mlir/utils/jenkins/Jenkinsfile.migraphxintegration

```diff
@@ -1,231 +1,317 @@
-          echo "(MLIR ENABLED) \$testcase \$datatype \$result" >> \$SUMMARY
-      done
-  done
+def createLogAndSummaryFiles(String arch) {
+    def logfile = "${arch}_generic.log"
+    def summary = "${arch}_summary.log"
+
+    sh """
+    echo "###########################################" >> ${logfile}
+    echo "New Run \$(pwd)" >> ${logfile}
+    date >> ${logfile}
+    echo "GPU: \$(/opt/rocm/bin/rocminfo | grep -o -m 1 'gfx.*')" >> ${logfile}
+    echo "MIGX: \$(\$(pwd)/AMDMIGraphX/build/bin/migraphx-driver --version)" >> ${logfile}
+    git config --global --add safe.directory \$(pwd)/AMDMIGraphX
+    echo "MIGX Commit: \$(git -C \$(pwd)/AMDMIGraphX log -n 1 --pretty=oneline)" >> ${logfile}
+    ls -l /etc/alternatives | grep "rocm ->" >> ${logfile}
+    echo "###########################################" >> ${logfile}
+    """
+}
+
+def testModels(modelsToTest, outLogFile, summary, datatypes) {
+    modelsToTest.each { testcase ->
+        if (testcase.startsWith('#')) {
+            return
+        }
+
+        datatypes.each { datatype ->
+            sh "echo 'Testing(MLIR ENABLED): ${testcase} ${datatype}' >> ${outLogFile}"
+            def command = "\$(pwd)/AMDMIGraphX/build/bin/migraphx-driver ${checkFor} ${testcase} ${datatype}"
+            def commandOutputFile = "command_output.txt"
+            def commandWithTee = "${command} | tee ${commandOutputFile}"
+            def timeoutStatus = sh(script: "timeout 1h ${commandWithTee}", returnStatus: true)
+            def output = readFile commandOutputFile
+
+            sh(script: "sed -n '/Summary:/,\$p; /FAILED:/,\$p' ${commandOutputFile} | tee -a ${outLogFile}")
+            def result = "DONE"
+
+            if (sh(script: "grep -q 'FAILED:' ${commandOutputFile}", returnStatus: true) == 0 ||
+                (sh(script: "grep -q 'Summary:' ${commandOutputFile}", returnStatus: true) != 0 && timeoutStatus != 124)) {
+                result = "FAILED"
+                error "Test case ${testcase} for datatype ${datatype} failed. Stopping execution."
+            } else if (timeoutStatus == 124) {
+                result = "TIMEOUT"
+            }
+            sh "echo '${result} Testing(MLIR ENABLED): ${testcase} ${datatype}\n' >> ${outLogFile}"
+            sh "echo '(MLIR ENABLED): ${testcase} ${datatype} ${result}' >> ${summary}"
+        }
+    }
 }
-rm -f tier1_p0.log
-rm -f tier1_p1.log
-rm -f other_models.log
-
-if [ "$enable_tier1_p0" = "true" ]; then
-	test_models tier1_p0_models tier1_p0.log
-fi
-if [ "$enable_tier1_p1" = "true" ]; then
-    test_models tier1_p1_models tier1_p1.log
-fi
-
-if [ "$enable_others" = "true" ]; then
-    test_models other_models other_models.log
-fi
-"""
+
+void buildAndTest(String arch) {
+    def datatypes = []
+    if (fp32 == "true") {
+        datatypes.add(" ")
+    }
+    if (fp16 == "true") {
+        datatypes.add("--fp16")
+    }
+    if (fp8 == "true") {
```

Comment:
Added testing for fp8. So far it has been successfully completed on navi2x, navi3x and mi300 machines.